src/Dfe.PlanTech.Web/Controllers/QuestionsController.cs

```diff
@@ -22,16 +23,23 @@ public class QuestionsController : BaseController<QuestionsController>
     /// <exception cref="ArgumentNullException">Throws exception when Id is null or empty</exception>
     /// <returns></returns>
     public async Task<IActionResult> GetQuestionById(
-        string id,
+        string? id,
         string? section,
-        int? submissionId,
-        string? answerRef,
         [FromServices] SubmitAnswerCommand submitAnswerCommand,
         CancellationToken cancellationToken)
     {
-        if (string.IsNullOrEmpty(id)) throw new ArgumentNullException(nameof(id));
-        object? parameters;
+        int? submissionId = null;
+        string? answerRef = null;
+
+        if (string.IsNullOrEmpty(id))
+        {
+            var parameterQuestionPage = DeserialiseParameter<TempDataQuestions>(TempData[TempDataConstants.Questions]);
+            id = parameterQuestionPage.QuestionRef;
+            submissionId = parameterQuestionPage.SubmissionId;
+            answerRef = parameterQuestionPage.AnswerRef;
+        }
 
+        object? parameters;
         TempData.TryGetValue("param", out parameters);
```

Comment:
This can be one lined, e.g. 

```
        TempData.TryGetValue("param", out object? parameters);
```

Reply:
Implemented in: [950beaa](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/216/commits/950beaacc13e623113783a5ccdef6dd31fb265b2)